### ISSUE TITLE
PrereleaseCompatibilityTransformer doesn't turn some messages into message objects

### ISSUE THREAD

**Issue by lgolding** [microsoft/sarif-sdk#1634]
Converting a 2.0.0 file to 2.1.0, the `PrereleaseCompatilityTransformer` leaves `reportingDescriptor.shortDescription` and `reportingDescriptor.fullDescription` as plain strings.

**Comment by lgolding** [comment]
According to the [Schema Changes document](https://github.com/oasis-tcs/sarif-spec/blob/master/Documents/Schema%20changes%20between%20SARIF%20v1%20and%20v2.md), `message` objects were first introduced in oasis-tcs/sarif-spec#84, which was closed on March 15, 2018. So only post-v1.0 SARIF files before that date would be in this condition, which is a narrow window.

According to the comments in PrereleaseCompatibilityTransformer.cs, the earliest changes it addresses are those accepted in TC meeting <span>#</span>25; that is, changes made _after_ TC <span>#</span>24, which occurred on September 25th, 2018.

In short, the PrereleaseCompatibilityTransformer doesn't aspire to transform versions earlier than that.
